app/main.py

```diff
@@ -1,13 +1,36 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"], "name": error["code"],
+            "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {"line": error["line_number"],
+             "column": error["column_number"],
+             "message": error["text"],
+             "name": error["code"], "source": "flake8"}
+            for error in errors
+        ],
+        "path": file_path,
+        "status": "failed",
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [
+                {"line": error["line_number"],
+                 "column": error["column_number"],
+                 "message": error["text"], "name": error["code"],
+                 "source": "flake8"}
+                for error in errors
+            ],
+            "path": file_path,
```

Comment:
The variable `file_path` is used here, but it is not defined within the scope of the `format_linter_report` function. Ensure that `file_path` is correctly defined or passed into the function.